client/components/basic/avatar/UserAvatarEditor.js

```diff
@@ -1,17 +1,30 @@
 import React, { useState, useCallback } from 'react';
-import { Box, Button, Icon, TextInput, Margins } from '@rocket.chat/fuselage';
+import { Box, Button, Icon, TextInput, Margins, Avatar } from '@rocket.chat/fuselage';
 
 import { useTranslation } from '../../../contexts/TranslationContext';
 import { useFileInput } from '../../../hooks/useFileInput';
-import UserAvatar from './UserAvatar';
+import { useAvatarUrlFromUserId } from '../../../hooks/useAvatarUrlFromUserId';
 
-export function UserAvatarEditor({ username, setAvatarObj }) {
+function UserAvatarSuggestions({ suggestions, setAvatarObj, setNewAvatarSource, disabled, ...props }) {
+	const handleClick = useCallback((suggestion) => () => {
+		setAvatarObj(suggestion);
+		setNewAvatarSource(suggestion.blob);
```

Comment:
If those setters are called together, it's interesting to trigger a single event for them, the reason being each setter call its own component update and we should let the parent component to decide if it two updates are needed or not (e.g. a single `dispatch` call from a `useReducer` hook).